app/chat/tasks.py

```diff
@@ -0,0 +1,119 @@
+from app.redis_service import RedisService
+from celery import app, group
+from celery.utils.log import get_task_logger
+from django.conf import settings
+from mattermostdriver import Driver
+
+logger = get_task_logger(__name__)
+
+redis = RedisService().redis
+
+# Lock timeout of 2 minutes (just in the case that the application hangs to avoid a redis deadlock)
+LOCK_TIMEOUT = 60 * 2
+
+chat_driver = Driver({
+    'url': settings.CHAT_URL,
+    'port': settings.CHAT_PORT,
+    'token': settings.CHAT_DRIVER_TOKEN
+})
+
+
+@app.shared_task(bind=True, max_retries=3)
+def create_channel(self, options, retry: bool = True) -> None:
+    """
+    :param options:
+    :param retry:
+    :return:
+    """
+    with redis.lock("tasks:create_channel:%s" % options['channel_name'], timeout=LOCK_TIMEOUT):
+
+        try:
+            chat_driver.login()
+            new_channel = chat_driver.channels.create_channel(options={
+                'team_id': options['team_id'],
+                'name': options['channel_name'],
+                'display_name': options['channel_display_name'],
+                'type': 'O'
+            })
+
+            return new_channel
+        except ConnectionError as exc:
+            logger.info(str(exc))
+            logger.info("Retrying connection")
+            self.retry(30)
+        except Exception as e:
+            print("we got an exception when creating a channel")
+            logger.error(str(e))
+
+
+@app.shared_task(bind=True, max_retries=3)
+def add_to_channel(self, options, retry: bool = True) -> None:
+    """
+    :param options:
+    :param retry:
+    :return:
+    """
+    with redis.lock("tasks:add_to_channel:%s" % options['channel_id'], timeout=LOCK_TIMEOUT):
+        chat_driver.login()
+        try:
+            for x in options['profiles']:
```

Comment:
What is x here? Please use variable names that make sense to increase readability.

Reply:
its a chat_id, will update

Reply:
@androolloyd you probably missed this ! Could you shoot a follow up PR?